### DISCUSSION TITLE
Error Starting GUI - "Current npm version(8.19.4) is less than 9. GUI app requires npm>=9." when on NPM:10.2.5 / Node:21.5.0

### DISCUSSION THREAD

**Discussion by BigPeteL** [Chia-Network/chia-blockchain#17108] [Support]
Hi, please can anyone assist with getting the GUI running on Ubuntu.  I'm getting an error "Current npm version(8.19.4) is less than 9. GUI app requires npm>=9."

npm 10.2.5 and node 21.5.0 are installed.
chia version: 2.1.3

Anyone have any suggestions?

TIA!

**Comment by BrandtH22** [comment]
Hey @BigPeteL , as a note we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia). 

As for the issues, if you are installing from source make sure to remove the venv and have it rebuilt to ensure the correct dependencies are being installed and used for chia. If you are installing from a packaged installer, what is your OS version?

**Reply by BigPeteL** [reply]
Yea I was upgrading from source.  Dumb question, how do I remove the venv?  (this is Ubuntu 22.04.3)

**Comment by BigPeteL** [comment]
FYI I've managed to get this working by renaming /chia-blockchain/.n to something else and the re-installing.  Now getting "Warning: Mismatched Versions"

Chia services version: 2.1.3
Chia application version: 1.3.6-dev157

**Reply by BrandtH22** [reply]
Hey @BigPeteL , as a note we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

Do you also have a packaged installer version of chia installed that could be causing the mismatch? (it sounds like there is a different executable being used that what is being installed via source)

If it is not a packaged installer version then I would remove that chia-blockchain folder that you had renamed and re git the source code then reinstall. Regarding the venv, during upgrades in the future I would run a `rm -R venv` while in the chia-blockchain directory and prior to running the sh install.sh command (this will remove the venv and all of its contents then the install will rebuild it with the latest dependencies)

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
